core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -88,6 +88,50 @@ class KryoSerializer(conf: SparkConf)
   private val useUnsafe = conf.get(KRYO_USE_UNSAFE)
   private val usePool = conf.get(KRYO_USE_POOL)
 
+  // classForName() is expensive in case the class is not found, so we filter the list of
+  // SQL / ML / MLlib classes once and then re-use that filtered list in newInstance() calls.
+  private lazy val loadableClasses: Seq[Class[_]] = {
```

Comment:
Could this be moved into a `private[serializer]` field in a `object KryoSerializer` companion? I'm worried that it'll be serialized as part of `KryoSerializer` itself, since I think the serializer itself is serialized as part of `ShuffleDependency`.

Reply:
Here's a commit with that change: https://github.com/joshrosen/spark/commit/c8680f9793775764b83a9735147376ffbe0b10a2

Reply:
Actually, I just pushed directly to your branch using GitHub's new "allow edits from maintainers" feature. Hope you don't mind 😄!

Reply:
Sure. Feel free to push it.